### PR TITLE
3p packages update 1.16.0

### LINKED ISSUE #3470: v1.16.0 update third party packages

### PR DESCRIPTION
**Pull request by vyaghras**
**Issue number:**

Closes #3470

**Description of changes:**
```
packages: update strace to 6.5
packages: update libtirpc to 1.3.4
packages: update libnl to 3.8.0
packages: update libglib to 2.78.0
packages: update kmod to 31
packages: update kexec-tools to 2.0.27
packages: update ethtool to 6.5
packages: update ecs-agent to 1.77.0
packages: update coreutils to 9.4
packages: update aws-signing-helper to 1.1.1
packages: update amazon-ssm-agent to 3.2.1630.0
```
Updates all third party packages except for the following:

binutils 2.38 -> 2.40 - bottlerocket-sdk has to move to 2.40 first.
docker related packages(docker-cli, docker-engine) 20.12.21 -> 20.12.23 - To sync with ECS-optimized AMI packaged docker version listed in [its version table](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-ami-versions.html#ecs-ami-versions-linux).
wicked 0.6.68 -> 0.6.70 - See update wicked to 0.6.70+ https://github.com/bottlerocket-os/bottlerocket/issues/2591
Containerd 1.6.24 -> 1.7.* - See https://github.com/bottlerocket-os/bottlerocket/issues/1963

**Tasks**
- [x] Upload packages in look aside cache
 
**Testing done:**

- [x] aws-k8s-1.24 x86_64
```
NAME                               TYPE               STATE                      PASSED          FAILED          SKIPPED   BUILD ID          LAST UPDATE
 x86-64-aws-k8s-124-quick           Test               passed                          1               0             6972   ca1cec36          2023-10-20T14:47:39Z
 x86-64-aws-k8s-124                 Resource           completed                                                            ca1cec36          2023-10-20T14:45:55Z
```
- [x] aws-k8s-1.24 aarch64
```
NAME                                TYPE               STATE                     PASSED          FAILED          SKIPPED   BUILD ID          LAST UPDATE
 aarch64-aws-k8s-124-quick           Test               passed                         1               0             6972   ca1cec36          2023-10-20T17:40:14Z
 aarch64-aws-k8s-124                 Resource           completed                                                           ca1cec36          2023-10-20T17:38:06Z
```
- [x] aws-k8s-1.24-nvidia x86_64
```
NAME                                             TYPE             STATE                  PASSED        FAILED        SKIPPED   BUILD ID        LAST UPDATE
 x86-64-aws-k8s-124-nvidia-quick                  Test             passed                      1             0           6972   c824540f        2023-10-20T23:59:01Z
 x86-64-aws-k8s-124-nvidia                        Resource         completed                                                    c824540f        2023-10-20T23:56:23Z
 x86-64-aws-k8s-124-nvidia-instances-rwsi         Resource         running                                                      c824540f        2023-10-20T23:59:13Z
```
- [x] aws-k8s-1.24-nvidia aarch64
```
NAME                                             TYPE            STATE                 PASSED       FAILED       SKIPPED   BUILD ID             LAST UPDATE
 aarch64-aws-k8s-124-nvidia-quick                 Test            passed                     1            0          6972   ca1cec36-dirty	 2023-10-20T19:18:10Z
 aarch64-aws-k8s-124-nvidia                       Resource        completed                                                 ca1cec36-dirty	 2023-10-20T19:15:01Z
 aarch64-aws-k8s-124-nvidia-instances-odlp        Resource        running                                                   ca1cec36-dirty	 2023-10-20T19:18:21Z
```
- [x] aws-k8s-1.27 x86_64
```
NAME                                       TYPE              STATE                   PASSED         FAILED         SKIPPED   BUILD ID         LAST UPDATE
 x86-64-aws-k8s-127-quick                   Test              passed                       5              0            7206   ca1cec36         2023-10-20T15:33:52Z
 x86-64-aws-k8s-127                         Resource          completed                                                       ca1cec36         2023-10-20T15:32:16Z
 x86-64-aws-k8s-127-instances-poby          Resource          running                                                         ca1cec36         2023-10-20T15:34:03Z
```
- [x] aws-k8s-1.27 aarch64
```
aarch64-aws-k8s-127-quick               Test             passed                       5              0            7206   ca1cec36-dirty        2023-10-20T18:16:34Z
 aarch64-aws-k8s-127                     Resource         completed                                                       ca1cec36-dirty        2023-10-20T18:14:31Z

```
- [x] aws-k8s-1.27-nvidia x86_64
```
x86-64-aws-k8s-127-nvidia-quick         Test             passed                       5              0            7206   ca1cec36-dirty        2023-10-20T22:03:15Z
 x86-64-aws-k8s-127-nvidia               Resource         completed                                                       ca1cec36-dirty        2023-10-20T22:00:08Z
```
- [x] aws-k8s-1.27-nvidia aarch64
```
 NAME                                             TYPE            STATE                 PASSED       FAILED	  SKIPPED   BUILD ID             LAST UPDATE
 aarch64-aws-k8s-127-nvidia-quick                 Test            passed                     5            0          7206   ca1cec36-dirty	 2023-10-20T21:29:21Z
 aarch64-aws-k8s-127-nvidia                       Resource        completed                                                 ca1cec36-dirty	 2023-10-20T21:26:43Z
 aarch64-aws-k8s-127-nvidia-instances-oimv        Resource        completed                                                 ca1cec36-dirty	 2023-10-20T21:26:54Z
```
- [x] aws-ecs-1 - Internal suite of tests passes successfully.
```
NAME                                     TYPE              STATE                     PASSED         FAILED         SKIPPED   BUILD ID         LAST UPDATE
 x86-64-aws-ecs-1-quick                   Test              passed                         1              0               0   c824540f         2023-10-20T22:57:15Z
 x86-64-aws-ecs-1                         Resource          completed                                                         c824540f         2023-10-20T22:56:42Z
 x86-64-aws-ecs-1-instances-dsyo          Resource          completed                                                         c824540f         2023-10-20T22:56:52Z


```
- [x] aws-ecs-1-nvidia
```
 NAME                                           TYPE             STATE                   PASSED         FAILED        SKIPPED   BUILD ID        LAST UPDATE
 x86-64-aws-ecs-1-nvidia-quick                  Test             passed                       1              0              0   c824540f        2023-10-20T23:11:26Z
```
- [x] aws-ecs-2
```
 x86-64-aws-ecs-2-quick                  Test             passed                       1              0               0   c824540f              2023-10-20T22:45:56Z
 x86-64-aws-ecs-2                        Resource         completed                                                       c824540f              2023-10-20T22:45:18Z
```
- [x] aws-ecs-2-nvidia
```
x86-64-aws-ecs-2-nvidia-quick                  Test             passed                       1              0              0   c824540f        2023-10-20T23:23:46Z
 x86-64-aws-ecs-2-nvidia                        Resource         completed                                                      c824540f        2023-10-20T23:22:58Z
 x86-64-aws-ecs-2-nvidia-instances-pmpe         Resource         running                                                        c824540f        2023-10-20T23:23:57Z
```
- [x] vmware-k8s-1.26
```
 NAME                                  TYPE               STATE                   PASSED          FAILED          SKIPPED   BUILD ID          LAST UPDATE
 x86-64-vmware-k8s-126-quick           Test               passed                       4               0             7068   c824540f          2023-10-23T21:19:40Z
 x86-64-vmware-k8s-126                 Resource           running                                                           c824540f          2023-10-23T21:20:00Z

```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: packages/libnl/Cargo.toml [hunk 1]
```diff
@@ -12,8 +12,8 @@ path = "../packages.rs"
 releases-url = "https://github.com/thom311/libnl/releases"
 
 [[package.metadata.build-package.external-files]]
-url = "https://github.com/thom311/libnl/archive/libnl3_7_0.tar.gz"
-sha512 = "86d9a5e9471495d29ab54377937dc662be693de40266d54c5b3690dbae627ed93b1ca9266b722d7e86c741e6f9ed6ea5129eb839e633f10ccc77e69dfd4f816a"
+url = "https://github.com/thom311/libnl/archive/libnl3_8_0.tar.gz"
+sha512 = "4e5e7187ecf6184ddfc38a95fdfe47efeb826f43f9f6546b5531780389ecf406f82f588275abea9970416b9fe8afa81527a7ddb771509f468ac5005774b0246c"
```
Comment (bcressey) [inline packages/libnl/Cargo.toml:16]: We should use the release artifacts here since those have published SHA256 hashes and a GPG signature.

**Review by bcressey — APPROVED** [review]

File: packages/libglib/Cargo.toml [hunk 2]
```diff
@@ -12,8 +12,8 @@ path = "../packages.rs"
 releases-url = "https://download.gnome.org/sources/glib"
 
 [[package.metadata.build-package.external-files]]
-url = "https://download.gnome.org/sources/glib/2.77/glib-2.77.2.tar.xz"
-sha512 = "0e2cf5178a7174b174480a795b1cbb2f1dbbea35899b7e4756937e426b6c39c20deb8488958403b78552674c4179e6fd63ea7fac2b746ac49a62046e27086111"
+url = "https://download.gnome.org/sources/glib/2.77/glib-2.77.3.tar.xz"
```
Comment (yeazelm) [inline packages/libglib/Cargo.toml]: There is a slightly newer version of `glib` if we want to take that as well: https://download.gnome.org/sources/glib/2.78/

**Review by yeazelm — APPROVED** [review]

File: packages/amazon-ssm-agent/amazon-ssm-agent.spec [hunk 3]
```diff
@@ -8,7 +8,7 @@
 %global goimport %{goproject}/%{gorepo}
 
 Name: %{_cross_os}amazon-ssm-agent
-Version: 3.2.1478.0
```
Comment (arnaldo2792) [inline packages/amazon-ssm-agent/amazon-ssm-agent.spec:11]: Let's use the version that the ECS agent uses today:

https://github.com/aws/amazon-ecs-agent/blob/master/CHANGELOG.md#1780

File: packages/ecs-agent/Cargo.toml [hunk 4]
```diff
@@ -13,8 +13,8 @@ path = "../packages.rs"
 releases-url = "https://github.com/aws/amazon-ecs-agent/releases"
 
 [[package.metadata.build-package.external-files]]
-url = "https://github.com/aws/amazon-ecs-agent/archive/v1.75.0/amazon-ecs-agent-1.75.0.tar.gz"
```
Comment (arnaldo2792) [inline packages/ecs-agent/Cargo.toml:16]: There is a newer version for the ECS agent.

**Review by arnaldo2792 — CHANGES_REQUESTED** [review]

File: packages/amazon-ssm-agent/amazon-ssm-agent.spec (hunk 3 above)
Comment (vyaghras) [reply packages/amazon-ssm-agent/amazon-ssm-agent.spec:11]: ECS-agent 1.77.0 uses very old version `3.1.804.0` that is not in sync with SSM agent version(3.2.1478.0) in latest Bottlerocket release. In ECS-agent [1.78.0](https://github.com/aws/amazon-ecs-agent/blob/master/CHANGELOG.md#1780) the SSM agent version will be `3.2.1630.0` but this  ECS-agent version has not been [released](https://github.com/aws/amazon-ecs-agent/releases) yet. So in this update we will keep SSM agent to `3.2.1630.0` version that will bring both of them in sync once we will update ECS agent to 1.78.0.

File: packages/ecs-agent/Cargo.toml (hunk 4 above)
Comment (vyaghras) [reply packages/ecs-agent/Cargo.toml:16]: I believe that is yet to [release](https://github.com/aws/amazon-ecs-agent/releases)

File: packages/libglib/Cargo.toml (hunk 2 above)
Comment (vyaghras) [reply packages/libglib/Cargo.toml]: Updated to latest.

**Comment by vyaghras** [comment]
⬆️  Upgraded ecs-agent to version 1.77.0

**Review by yeazelm — APPROVED** [review]

**Review by arnaldo2792 — APPROVED** [review]

**Pull request merged**